services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,47 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if(event == NULL) return ERR_CODE_INVALID_ARG;
+
+  if(xQueueSend(thermalMgrQueueHandle, (void *) event, TEMPERATURE_WAIT_TIME) != pdTRUE) {
+    return ERR_CODE_QUEUE_FULL;
+  }
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t event = {.type = THERMAL_OS_HANDLE};
+  BaseType_t xHigherPriorityTaskWoken = pdFALSE;
+  while(1) {
+   if(xQueueSendFromISR(thermalMgrQueueHandle, (void *) &event, &xHigherPriorityTaskWoken) == pdTRUE)
+    break;
+  }
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
+  // Set variables and get config for lm75bd
+  thermal_mgr_event_t event;
+  float temp;
+  lm75bd_config_t *config = (lm75bd_config_t *) pvParameters;
+  
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, (void *) &event, TEMPERATURE_WAIT_TIME) == pdTRUE) {
+      readTempLM75BD(config->devAddr, &temp);
+
+      // Part 2 of challenge
+      if (event.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD){
+        addTemperatureTelemetry(temp);
+
+      // Part 3 of challenge 
+      } else if(event.type == THERMAL_OS_HANDLE) {
+        if(temp > config->hysteresisThresholdCelsius) {
+          overTemperatureDetected();
+        } else {
+          safeOperatingConditions();
+        }
+      }
+    }
```

Comment:
Handle the case where an invalid event is received (i.e. `.type` is an incorrect value)